crates/core/src/storage/file.rs

```diff
@@ -375,6 +472,10 @@ mod imp {
         }
     }
 
+    pub(super) async fn regular_rename(from: &str, to: &str) -> Result<(), LocalFileSystemError> {
+        rename_noreplace(from, to).await
```

Comment:
Why is this calling rename_noreplace here?

Reply:
For macOS, platform_specific_rename function will be invoked instead of utilizing hard links, as macOS does not support fuse file systems. Therefore, it's advisable to use rename_noreplace in this context.